drivers/hdc1000/hdc1000.c

```diff
@@ -110,21 +119,64 @@ int hdc1000_get_results(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
     i2c_release(dev->p.i2c);
 
     if (status == HDC1000_OK) {
+        /* update last_read_time */
+        dev->last_read_time = xtimer_now_usec();
+
         /* if all ok, we convert the values to their physical representation */
+        uint16_t traw = ((uint16_t)buf[0] << 8) | buf[1];
+        dev->temp_cached = (int16_t)((((int32_t)traw * 16500) >> 16) - 4000);
+
+        uint16_t hraw = ((uint16_t)buf[2] << 8) | buf[3];
+        dev->hum_cached  = (int16_t)(((int32_t)hraw * 10000) >> 16);
+
         if (temp) {
-            uint16_t traw = ((uint16_t)buf[0] << 8) | buf[1];
-            *temp = (int16_t)((((int32_t)traw * 16500) >> 16) - 4000);
+            *temp = dev->temp_cached;
         }
         if (hum) {
-            uint16_t hraw = ((uint16_t)buf[2] << 8) | buf[3];
-            *hum  = (int16_t)(((int32_t)hraw * 10000) >> 16);
+            *hum = dev->hum_cached;
         }
     }
 
     return status;
 }
 
-int hdc1000_read(const hdc1000_t *dev, int16_t *temp, int16_t *hum)
+bool hdc1000_outdated(hdc1000_t *dev)
+{
+    /* initialization phase */
+    if (dev->last_read_time == 0xFFFFFFFF && dev->hum_cached == 0 && dev->temp_cached == 0) {
+        return true;
+    }
+
+    uint32_t now = xtimer_now_usec();
+    if (now >= dev->last_read_time) {
+        if (now - dev->last_read_time > HDC1000_RENEW_INTERVAL) {
+            return true;
+        }
+    }
+    else {
+        if (0xFFFFFFFF - dev->last_read_time + now > HDC1000_RENEW_INTERVAL) {
+            return true;
+        }
+    }
+    return false;
+}
```

Comment:
Correct me if I'm wrong, but with my statement above, the whole function could be simplified to:

```
if((now - last) > INTERVAL) true
else false
```

This is pretty simple so I'd like to keep it sleep and abstain from introducing the extra function `hdc1000_outdated`. Simply implement this in `hdc1000_read_cached`